starlette_admin/contrib/beanie/view.py

```diff
@@ -0,0 +1,280 @@
+import functools
+from typing import (
+    Any,
+    Dict,
+    Generic,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    get_args,
+    get_origin,
+)
+
+import bson.errors
+import starlette_admin.fields as sa
+from beanie import Document, Link, PydanticObjectId
+from mongoengine.queryset import QNode
+from mongoengine.queryset.visitor import QCombination
+from pydantic import ValidationError
+from starlette.requests import Request
+from starlette_admin._types import RequestAction
+from starlette_admin.contrib.beanie.converters import (
+    BeanieModelConverter,
+)
+from starlette_admin.contrib.beanie.helpers import (
+    Q,
+    build_order_clauses,
+    flatten_qcombination,
+    is_link_type,
+    is_list_of_links_type,
+    resolve_deep_query,
+)
+from starlette_admin.helpers import (
+    not_none,
+    prettify_class_name,
+    pydantic_error_to_form_validation_errors,
+    slugify_class_name,
+)
+from starlette_admin.views import BaseModelView
+
+T = TypeVar("T", bound=Document)
+
+
+class ModelView(BaseModelView, Generic[T]):
+    full_text_override_order_by: bool = False
+
+    def __init__(
+        self,
+        document: Type[T],
+        icon: Optional[str] = None,
+        name: Optional[str] = None,
+        label: Optional[str] = None,
+        identity: Optional[str] = None,
+        converter: Optional[BeanieModelConverter] = None,
+    ):
+        self.document = document
+        self.identity = (
+            identity or self.identity or slugify_class_name(self.document.__name__)
+        )
+        self.label = (
+            label or self.label or prettify_class_name(self.document.__name__) + "s"
+        )
+        self.name = name or self.name or prettify_class_name(self.document.__name__)
+        self.icon = icon
+        self.pk_attr = "id"
+        self.has_full_text_index: Optional[bool] = None
+
+        self.fields_pydantic = list(document.model_fields.items())
+
+        self.field_infos = []
+        self.link_fields = []
+        self.exclude_fields_from_create = [
```

Comment:
Can you also add a function that will converts a  list like `[Product.name, Product.description]` into `["name", "description"]` https://github.com/jowilf/starlette-admin/blob/fc995cbbbc797fc37e172829f6c359e81dbbdb0e/starlette_admin/contrib/odmantic/view.py#L67